packages/frontend2/src/server/features/scaling/get-scaling-liveness-entries.tsx

```diff
@@ -0,0 +1,137 @@
+import { type Layer2, layer2s } from '@l2beat/config'
+import {
+  TrackedTxsConfigSubtypeValues,
+  UnixTime,
+  notUndefined,
+} from '@l2beat/shared-pure'
+import { getImplementationChangeReport } from '../implementation-change-report/get-implementation-change-report'
+import { toAnomalyIndicatorEntries } from '../liveness/get-anomaly-entries'
+import { getLiveness } from '../liveness/get-liveness'
+import { type LivenessProject } from '../liveness/types'
+
+import { getProjectsVerificationStatuses } from '../verification-status/get-projects-verification-statuses'
+import { getCommonScalingEntry } from './get-common-scaling-entry'
+import { getLatestTvlUsd } from './tvl/utils/get-latest-tvl-usd'
+import { orderByTvl } from './tvl/utils/order-by-tvl'
+
+export async function getScalingLivenessEntries() {
+  const [
+    tvl,
+    projectsVerificationStatuses,
+    implementationChangeReport,
+    liveness,
+  ] = await Promise.all([
+    getLatestTvlUsd(),
+    getProjectsVerificationStatuses(),
+    getImplementationChangeReport(),
+    getLiveness(),
+  ])
+
+  const activeProjects = layer2s.filter(
+    (p) =>
+      liveness[p.id.toString()] &&
+      (p.display.category === 'Optimistic Rollup' ||
+        p.display.category === 'ZK Rollup') &&
+      !p.isUpcoming &&
+      !p.isArchived,
+  )
```

Comment:
I've seen we have `getLivenessProjects` somewhere around - how about injecting liveness into this function or something like this?

Reply:
good catch